ouroboros-consensus/src/Data/SOP/Strict.hs

```diff
@@ -60,6 +62,19 @@ type instance AllZipN    NP c = AllZip c
 type instance Prod       NP   = NP
 type instance SListIN    NP   = SListI
 
+instance forall k xs . (Typeable (NP k xs), All (Compose ToCBOR k) xs) => ToCBOR (NP k xs) where
+  toCBOR xsTop = go xsTop
```

Comment:
My `nix` is currently rebuilding the world :/, so I'll just ask what you think: can we use `hcfoldMap` here instead of this `go`?